jena-arq/src/main/java/org/apache/jena/sparql/core/mosaic/LockTry.java

```diff
@@ -0,0 +1,8 @@
+package org.apache.jena.sparql.core.mosaic;
+
+import org.apache.jena.shared.Lock;
+
+public interface LockTry extends Lock {
+
+	boolean tryEnterCriticalSection(boolean readLockRequested);
```

Comment:
It might be nice to put a Javadoc comment on this to make the required semantics precisely known. They seem clear now, but as Jena's concurrency evolves, they might get less so.